presto-main/src/main/java/com/facebook/presto/execution/buffer/AESSpillCipher.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.buffer;
+
+import com.facebook.presto.spi.PrestoException;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+final class AESSpillCipher
+        implements SpillCipher
+{
+    public enum Mode
+    {
+        AES_128(128),
+        AES_256(256);
+
+        private final int keyBits;
+
+        Mode(int keyBits)
+        {
+            this.keyBits = keyBits;
+        }
+
+        public int getKeyBits()
+        {
+            return keyBits;
+        }
+    }
+
+    private SecretKeySpec key;
+    private final int keyBits;
+    private final int ivBytes;
+
+    AESSpillCipher(Mode mode)
+    {
+        this.keyBits = requireNonNull(mode, "mode is null").getKeyBits();
+        this.key = generateNewSecretKey(keyBits);
+        this.ivBytes = createEncryptCipher().getIV().length;
+    }
+
+    @Override
+    public ByteBuffer decrypt(ByteBuffer cipherText)
+    {
+        int length = cipherText.remaining();
+        if (length < ivBytes) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, format("cipherText too small to contain IV. Minimum: %s, Actual: %s", ivBytes, length));
```

Comment:
Perhaps, clarify the error message and spell out IV: `Encrypted data size is too small: %s. It must be at least the size of the initialization vector: %s.`